engines/python/setup/djl_python/streaming_utils.py

```diff
@@ -32,62 +35,105 @@ def get_stream_generator(execution_engine: str):
+                                        past_key_values=past_key_values,
+                                        use_cache=True)
+
+            if generic_model_class == "Seq2SeqLM":
+                outputs = model.forward(
+                    input_ids=input_ids,
+                    attention_mask=attention_mask,
+                    decoder_input_ids=decoder_input_ids,
+                    decoder_attention_mask=decoder_attention_mask,
+                    encoder_outputs=encoder_last_hidden_state,
+                    past_key_values=past_key_values,
+                    use_cache=True)
+
             next_token_ids = []
+            ## TODO: batch decoding
             for i, logits in enumerate(outputs.logits):
                 next_token_id = decoding_method(
-                    logits, all_input_ids[i, :].view(1, -1), **kwargs)
+                    logits, all_decoder_input_ids[i, :].view(1, -1), **kwargs)
                 next_token_ids.append(next_token_id.view(1, 1))
             token_ids = torch.cat(next_token_ids)
+
+            all_decoder_input_ids = torch.cat(
+                [all_decoder_input_ids, token_ids], dim=1)
             past_key_values = outputs.past_key_values
-            attention_mask[:, curr_length] = 1
-            curr_length += 1
             new_tokens_count += 1
 
             not_eos_token_ids = (token_ids != tokenizer.eos_token_id).view(
                 len(inputs), 1)
             unfinished_sequences = unfinished_sequences.mul(not_eos_token_ids)
 
-            input_ids = token_ids.view(len(inputs), 1)
-            input_ids = input_ids * unfinished_sequences + tokenizer.pad_token_id * unfinished_sequences.logical_not(
-            )
-            all_input_ids = torch.cat([all_input_ids, token_ids], dim=1)
-            token_text = tokenizer.batch_decode(input_ids)
+            if generic_model_class == "CausalLM":
+                input_ids = token_ids.view(len(inputs), 1)
+                input_ids = input_ids * unfinished_sequences + tokenizer.pad_token_id * unfinished_sequences.logical_not(
+                )
+                token_text = tokenizer.batch_decode(input_ids)
+                attention_mask[:, curr_length] = 1
+                curr_length += 1
+
+            if generic_model_class == "Seq2SeqLM":
+                input_ids = None
+                decoder_input_ids = token_ids.view(len(inputs), 1)
+                decoder_input_ids = decoder_input_ids * unfinished_sequences + tokenizer.pad_token_id * unfinished_sequences.logical_not(
+                )
+                encoder_last_hidden_state = [outputs.encoder_last_hidden_state]
+                token_text = tokenizer.batch_decode(decoder_input_ids)
 
             stop_generation = StreamingUtils._has_met_stopping_criteria(
-                not_eos_token_ids, new_tokens_count, max_new_tokens)
+                unfinished_sequences, new_tokens_count, max_new_tokens)
+
+            if stop_generation:
```

Comment:
will we miss the last token from the tokenizer?

Reply:
good catch, I bought this check to the bottom to prevent sending end of sentence token if that's what is causing the generation to stop but it is incorrect as it will prevent sending last token if max new tokens condition is met. Reverted the change.